// sanity-plugin-media/dist/app.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var ui_1 = require("@sanity/ui");
var react_1 = __importDefault(require("react"));
var theme_ui_1 = require("theme-ui");
var constants_1 = require("./constants");
var AssetSourceDispatchContext_1 = require("./contexts/AssetSourceDispatchContext");
var Browser_1 = __importDefault(require("./components/Browser"));
var ReduxProvider_1 = __importDefault(require("./components/ReduxProvider"));
var useKeyPress_1 = __importDefault(require("./hooks/useKeyPress"));
var GlobalStyles_1 = __importDefault(require("./styled/GlobalStyles"));
var theme_1 = __importDefault(require("./styled/theme"));
var AssetBrowser = function (props) {
    var onClose = props.onClose, onSelect = props.onSelect, tool = props.tool;
    // Close on escape key press
    useKeyPress_1.default('escape', onClose);
    // Stop propagation and prevent document mouse events from firing.
    // This is a bit of a hack to make this work with `editModal = 'popover'` and prevent Sanity's <Popover /> component from
    // prematurely closing, as it attaches events on `document` to detect outside clicks.
    var handleStopPropagation = function (e) {
        e.nativeEvent.stopImmediatePropagation();
        e.stopPropagation();
    };
    return (react_1.default.createElement(ReduxProvider_1.default, __assign({}, props),
        react_1.default.createElement(ui_1.ThemeProvider, { scheme: "dark", theme: ui_1.studioTheme },
            react_1.default.createElement(theme_ui_1.ThemeProvider, { theme: theme_1.default },
                react_1.default.createElement(ui_1.PortalProvider, { element: document.body },
                    react_1.default.createElement(ui_1.ToastProvider, { zOffset: constants_1.Z_INDEX_TOAST_PROVIDER },
                        react_1.default.createElement(AssetSourceDispatchContext_1.AssetBrowserDispatchProvider, { onSelect: onSelect },
                            react_1.default.createElement(GlobalStyles_1.default, null),
                            tool ? (react_1.default.createElement(ui_1.Box, { style: { height: '100%', position: 'relative' } },
                                react_1.default.createElement(Browser_1.default, { onClose: onClose }))) : (react_1.default.createElement(ui_1.Portal, null,
                                react_1.default.createElement(ui_1.Box, { onDragEnter: handleStopPropagation, onDragLeave: handleStopPropagation, onDragOver: handleStopPropagation, onDrop: handleStopPropagation, onMouseUp: handleStopPropagation, style: {
                                        bottom: 0,
                                        height: 'auto',
                                        left: 0,
                                        position: 'fixed',
                                        top: 0,
                                        width: '100%',
                                        zIndex: constants_1.Z_INDEX_APP
                                    } },
                                    react_1.default.createElement(Browser_1.default, { onClose: onClose })))))))))));
};
exports.default = AssetBrowser;
